specification/storage/resource-manager/Microsoft.Storage/preview/2020-08-01-preview/storage.json

```diff
@@ -2132,6 +2135,28 @@
         }
       }
     },
+    "EncryptionIdentity":{
+       "description": "Identity to be used for server-side encryption on the storage account",
+       "properties": {
+         "userAssignedIdentity":{
+           "type": "string",
+           "description": "ResourceID of the UserAssignedIdentity to be associated with server-side encryption on the storage account.",
+           "x-ms-client-name": "UserAssignedIdentity"
+         },
+         "type": {
+          "type": "string",
+          "description": "The Encryption Identity type.",
+          "enum": [
+            "ManagedServiceIdentity",
+			      "StorageService"
```

Comment:
does only one of ManagedServiceIdentity, StorageService can be specified?
If not specify any in create account, what's the default value?

Besides that, if we add "userAssignedIdentity", without specify this type, or specify "StorageService", what will happen?
If we don't add "userAssignedIdentity", and specify "ManagedServiceIdentity" what will happen?

It seems the value should be decided from if we add "userAssignedIdentity". If so, why we need this property?

Reply:
Does only one of "ManagedServiceIdentity", "StorageService" can be specified?
Yes - only one of the 2. As of now this property defaults to "ManagedServiceIdentity" and we limit who has permissions to use "StorageService" - internal partners typically. So there is no requirement to support toggling this in the client tooling.

Reply:
If not specify any in create account, what's the default value?
The default value is "ManagedServiceIdentity"

Reply:
Besides that, if we add "userAssignedIdentity", without specify this type, or specify "StorageService", what will happen?
If we don't add "userAssignedIdentity", and specify "ManagedServiceIdentity" what will happen?

- If you specify type = StorageService - unless you are an internal service we permit, this request will be rejected. Regardless of what is set in userAssignedIdentity.
- type = ManagedServiceIdentity is the default and is what is set when userAssignedIdentity is set. 
- if userAssignedIdentity is set without type being specified, the type defaults to ManagedServiceIdentity.

Reply:
It seems the value should be decided from if we add "userAssignedIdentity". If so, why we need this property?
Because internal partners can toggle this to "StorageService" which allows them to bypass using MSIs for CMK. Having said this, there is no requirement to permit setting this property via client tooling.